### ISSUE TITLE
[BUG] 360 video is flattened

### ISSUE THREAD

**Issue by JesseWebDotCom** [immich-app/immich#5414]
### The bug

When re-encoding 360 video, immich flattens the video.

To confirm:
- take a 360 degree mp4 video, open in VLC, and confirm you can properly rotate the video in 360 degrees
- now have immich re-encode that video (HEVC in my case) and do the same test in VLC - you will not be able to rotate the video

My theory is immich loses the 360 side data when using FFmpeg to re-encode. For example, here is some ffprobe side data for a 360 video that I believe immich loses:
```
  Stream #0:0[0x1](und): Video: hevc (Main) (hvc1 / 0x31637668), yuv420p(tv, bt709), 4096x2048 [SAR 1:1 DAR 2:1], 28334 kb/s, 29.97 fps, 29.97 tbr, 90k tbn (default)
    Metadata:
      creation_time   : 2022-07-27T13:27:56.000000Z
      handler_name    : VideoHandler
      vendor_id       : [0][0][0][0]
    Side data:
      spherical: equirectangular (0.000000/0.000000/0.000000) 
```

When re-encoding, immich needs to add -strict unofficial flag to the ffmpeg command line in order to copy the video projection metadata.  See here for more detail: https://stackoverflow.com/questions/44760588/preserving-side-data-information-for-360-video-transcoding-using-ffmpeg/48147865#48147865

### The OS that Immich Server is running on

Ubuntu 22.10

### Version of Immich Server

v1.89.0

### Version of Immich Mobile App

v1.89.0

### Platform with the issue

- [X] Server
- [X] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
n/a
```

### Your .env content

```Shell
n/a
```

### Reproduction steps

```bash
see above
```

**Comment by johmicrot** [comment]
I'm on the latest release (v.1.111.0) and this is still a persisting issue

**Comment by mertalev** [comment]
Sorry, we have to revert the PR for this for now. It had the side effect of breaking playback when the source video uses Dolby Vision. The transcode ends up having incompatible DOVI metadata that players try and fail to use. It's surprisingly difficult to remove this and *just* this metadata, but will be easier in FFmpeg 7.1.

**Comment by Dunky13** [comment]
I have a whole batch of Garmin 360 video & photos. This would be a great feature for me 🙏

**Comment by Rambomst** [comment]
I recently purchased an Insta360 x4 and came across this issue, at least I can view the front camera view but it would be great if this was implemented.

**Comment by jarrah31** [comment]
I too would like this feature so that I can watch 360 videos from my Insta360 X5 on the Immich mobile app and web views.

**Comment by tienli** [comment]
This probably is something need to be fixed (or implemented) from photo side up.
Use a 360 picture and save it in either Google photo or Immich server. I can see 360 degree photo in browser both in Google and Immich, while doing the same in apps will only see 360 degree photo in Google's app; not in Immich's mobile app (I used iPhone, yet tried Android).  It makes me thinking that rendering is without a problem. Since browser can do it, there must be something missing or wrong where app implemented. Hopefully I'd remember to upload a 360-degree picture here later so anyone can try it out.
Should I open a new issue?

**Comment by danieldietzler** [comment]
> Should I open a new issue?

No, why should you? There's no need to duplicate an existing issue lol

**Comment by 1661410107114145152525** [comment]
> Since browser can do it, there must be something missing or wrong where app implemented. Hopefully I'd remember to upload a 360-degree picture here later so anyone can try it out. Should I open a new issue?

@tienli Immich intentionally doesn't support 360° or panorama content at all on mobile. As there will propably never be an open source library for mobile 360° / panorama content because browser libraries became more and more popular, we might never see something in the app. Only other way would be that Immich integrates an embedded browser view for such content, where the photoshpere plugin that the web version uses, is used. But this is another issue / festure request...

**Comment by mertalev** [comment]
I'm sure there is an API or library to do it on mobile too, but it hasn't been implemented yet.

**Comment by dlang123** [comment]
I too want 360 video on both web and mobile, have an insta360 x5 and x4 video that I would love to store in Immich too since it is family vacations.

**Comment by george1612** [comment]
@dlang123 Found this issue https://github.com/immich-app/immich/discussions/26692

**Comment by meesfrensel** [comment]
I think the original issue's _user expectation_ has been fixed long ago with the introduction of the panorama video viewer in the web client. For the mobile client there is #6572, and for the transcoded videos, this is not a bug (and inside Immich, the transcoded video isn't even used for 360 display).
